unit-tests/src/test/java/org/eclipse/collections/impl/bag/mutable/MutableBagTestCase.java

```diff
@@ -450,7 +450,7 @@ public void selectDuplicates()
     @Test
     public void topOccurrences()
     {
-        MutableBagIterable<String> strings = this.newWithOccurrences(
+        MutableBagIterable<String> strings = Bags.mutable.withOccurrences(
```

Comment:
Please add at least one test where the number of occurrences is Integer.MAX_VALUE. This should fail with OutOfMemoryError until the pre-sizing fix is in place.

Reply:
I tried to change the last element of occurrences to `Integer.MAX_VALUE`. I didn't meet OutOfMemoryError even I didn't fix pre-sizing. 
Here is the change: 
```java
MutableBagIterable<String> strings = Bags.mutable.withOccurrences(
                PrimitiveTuples.pair("one", 1),
                PrimitiveTuples.pair("two", 2),
                PrimitiveTuples.pair("three", 3),
                PrimitiveTuples.pair("four", 4),
                PrimitiveTuples.pair("five", 5),
                PrimitiveTuples.pair("six", 6),
                PrimitiveTuples.pair("seven", 7),
                PrimitiveTuples.pair("eight", 8),
                PrimitiveTuples.pair("nine", 9),
                PrimitiveTuples.pair("ten", Integer.MAX_VALUE));
```

Should I still add one test for this? 

But I do understand that the reason of pre-sizing for the variable arguments. The underlying storage could grow dynamically and exceed than we need. So it will be better to pre-sizing in the initialization.